consensus/src/consensus/consensus_proxy.rs

```diff
@@ -32,14 +33,27 @@ use tokio_stream::wrappers::BroadcastStream;
 
 use super::{ConsensusRequest, ResolveBlockError, ResolveBlockRequest};
 use crate::{
-    consensus::remote_data_store::RemoteDataStore,
+    consensus::{remote_data_store::RemoteDataStore, SyncProgress},
     messages::{
         AddressNotification, AddressSubscriptionOperation, AddressSubscriptionTopic,
         RequestBlocksProof, RequestSubscribeToAddress, RequestTransactionReceiptsByAddress,
         RequestTransactionsProof, ResponseBlocksProof,
     },
     ConsensusEvent,
 };
+
+/// This struct is used to track the progress of the consensus sync
+/// and return it to the user.
+#[derive(Debug, Clone, Serialize, Deserialize, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct ConsensusSyncStatus {
+    pub is_established: bool,
+    pub synced_validity_window: bool,
+    pub current_block: u32,
+    pub remaning_blocks: u32,
```

Comment:
typo, please refactor
```suggestion
    pub remaining_blocks: u32,
```